app/src/Containers/JokeForm/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from "react"
+import { JokeInput, JokeWrapper, SubmitButton } from "./styles"
+import { useDispatch, useSelector } from "react-redux"
+import { addJoke } from "../../actions"
+
+const JokeForm = () => {
+ const [jokeInput,setJoke] = useState("")
+ const dispatch = useDispatch()
+ const joke = useSelector(state => state.joke)
+ const [error,setError] = useState(joke.error)
+
+ const handleInput = (event: React.FormEvent<HTMLInputElement>) => {
+   setJoke(event.currentTarget.value)
+ }
+
+ const handleJokeCreation = () => {
+   const new_joke = {
+     new_joke: jokeInput
+   }
```

Comment:
this is the line that makes this whole thing hard for me to reason about. the reference by environment is nat a good look here. it's an approach that works but there are quite a few side effects that happen by referencing the same variable everywhere, and thus is discouraged. let's talk about ideal state:

```tsx
const handleJokeCreation = (jokeInput: string) => {
  if (jokeInput.length === 0) {
    setError("enter a joke")
    return setTimeout(() => setError(""), 6000)
  }
  return dispatch(addJoke(jokeInput))
}
```

and then it should be invoked like this:

```tsx
<SubmitButton disabled={joke.isLoading} onClick={() => handleJokeCreation(jokeInput)}>{joke.isLoading ? "Sending" : "Submit Joke"}</SubmitButton>
```

this clarifies who the actor is, where the value comes from, how it gets set, and doesn't attempt to create a payload for a netowrk request that's three or four function calls from invocation. when you put it all together here you had a sharted reference to a variable, that calls a dispatch and passes to it an action that requires a json payload for a network request to flow through to the promise that it returns.

it's hard to follow